### DISCUSSION TITLE
How would I go about making multiple Super forms?

### DISCUSSION THREAD

**Discussion by Fritzmeinzer** [Techokami/SonicWorldsNext#157] [Q&A] [answered]
So I was just looking through the overall engine and the color palletes and I wanted to change some things so there's multiple Super forms. I was intending on having Super, Hyper, and Miracle Sonic (non canon Sonic CD super form).

Should I just duplicate the SuperStart state and change it to HyperStart and MiracleStart respectfully? Additionally, what should I do in terms of replacing the sprites and palletes? Is there a way where I can just find a similar sprite sheet where all the other sprites were ripped or just replicate all the animations by cutting and pasting in aseprite?

Thanks in advance

**Comment by Renhoex** [comment, marked as answer]
You're probably going to have to make a variable to check what type of form the players in like `superForm = 0`
you don't need to make new state's for activating super forms, you'd just have to find where the super start state is and add what superForm you want to use there.

For the palettes, check the player graphic scene, there's an animator for `SuperPalette` you'd just need to add a track for each of the form cycling animations, to modify the palette cycling you can check the shader, if you want more colours you'll have to expand the colours to include the palettes for the other forms (keep in mind this should be to the power of 2 for the best result)
![image](https://github.com/Techokami/SonicWorldsNext/assets/26295900/f7373fbc-074e-47c0-a1a7-13843d81fd83)

as for new sheets, yes unfortunately you're going to create a copy of the sprite sheet you're using then copy and paste the frames into it, using asesprite I'd recommend doing an import so you can set it up as an animation sheet and have an easier time aligning the sprites.

This process can be very involved so I'd recommend only doing this if it's completely necessary to what you'd like to accomplish.

**Reply by Fritzmeinzer** [reply]
Im trying to update the engine to give it more features and a look similar to the CORE Engine with Mania sprites. I achieved the different forms by copy pasting the Super Animator.  I didn't try the Superform variable and  I used isHyper and isMiracle. But, I'll use that variable as well.

Also, while not necessarily on topic is there a way I can incorparate Time travel between Past, Present and Good/Bad Future? Part of Miracle's gimmick is Peelout for Future and Spindash for Past

**Reply by r6915ee** [reply]
There was [this](https://github.com/Techokami/SonicWorldsNext/discussions/42#discussion-5230529) discussion at one point that might be able to help you out in some way for time travel.
